lib/cWarnings.ml

```diff
@@ -152,10 +152,75 @@ let parse_flags s =
 let set_flags s =
   reset_default_warnings (); let s = parse_flags s in flags := s
 
+type category =
+  | Automation
+  | Bytecode_compiler
+  | Debug
+  | Deprecated
+  | Dev
+  | Extraction
+  | Filesystem
+  | Fixpoints
+  | Funind
+  | Implicits
+  | Loadpath
+  | Ltac
+  | Native_compiler
+  | Non_interactive
+  | Notation
+  | Numbers
+  | Option
+  | Parsing
+  | Pedantic
+  | Records
+  | Require
+  | Schemes
+  | Scope
+  | Ssr
+  | Syntax
+  | Tactics
+  | Typechecker
+  | Typeclasses
+  | Vernacular
+  | Other of string
```

Comment:
That's nice [IMO] ; instead of other you could make this type user-extensible using the `+=` syntax.

Reply:
Can you be a bit more specific or point to the doc? I don't know this syntax. Does this amount to the following change?
```suggestion
  +=
```

Reply:
https://caml.inria.fr/pub/docs/manual-ocaml/extensiblevariants.html

Reply:
To use them properly adds a bit of overhead as the best alternative is to use the private extensible extension as thus provide the right API for that; I'd like to something similar for errors tho, so it could be interesting to do a trial here.

Reply:
Thanks! Unfortunately this won't work because we need to be able to produce a string from the constructor name in `category_to_string`, and pattern-matching on extensible variants would only allow one to return a default value for unknown constructors.

Reply:
> private extensible extension

AFAIU it is only supported in OCaml 4.06+. (I also don't see how you would use them, so I'll leave it to you to refine this in a further PR.)

Reply:
The way it works is as describe in the documentation, you extend using a functor which also provides the printing function; you can still do the same in 4.05 but just dropping the `private` keyword, this won't enforce statically that people properly register constructors but otherwise the API would be the right one so IMO it is much preferred to have this right from the start.

Reply:
OK, I'll look closer into this.

Reply:
I can help and provide example code, if you give a bit of time.

Reply:
Will do, there's no rush to get this in.

Reply:
Here's some example code that does this https://github.com/coq/coq/pull/12602/files#diff-bec0b7d135d40ef82bf372651ab32815ab340fe61bc75b9d9ae7c6ee1b362aaeR47